### ISSUE TITLE
Run corebird in the background

### ISSUE THREAD

**Issue by haecker-felix** [baedert/corebird#702]
It would be nice, if I could close the corebird main window, but get still notifications about new tweets etc.

**Comment by Vistaus** [comment]
A thousand times yes! +1

**Comment by baedert** [comment]
You can just run `corebird --start-service` but nobody ever makes sure that it keeps working.

**Comment by Vistaus** [comment]
@baedert Great tip! I could add that to my startup applications. But it shouldn't be hard to write a daemon around that, right? (note that I don't mean to suggest that you should do it, just asking a general question)

**Comment by baedert** [comment]
what kind of daemon? It'll only start the streaming connection without any of the UI this way, so you will get the usual notifications and starting corebird will just create the UI. Closing the window will destroy the UI but leave the process running until you execute `corebird --stop-service`.

**Comment by Vistaus** [comment]
In that case it'd be nice to have this available as a GUI option :) I mean: not all users are going to search GitHub Issues for this.

**Comment by baedert** [comment]
Yeah well it's not really supposed to be used :p It's even harder to test that this keeps working (and it requires a certain decoupled-from-the-actual-UI design) than the usual stuff. Additionally I'd like he close button to somehow indicate that the process keeps running but I'm not sure how to best implement this.

**Comment by corebots** [comment]
just a hint, closing the window could make the nice icon hi-contrasted appear on the top gnome-shell panel indicating that the service is running in the background

**Comment by baedert** [comment]
>just a hint, closing the window could make the nice icon hi-contrasted appear on the top gnome-shell panel indicating that the service is running in the background

That would have to be a gnome-shell extension then and doesn't work anywhere else.

**Comment by Vistaus** [comment]
@baedert Well, starting the service upon boot does work and seems to be running all of the time. Only issue is that notifications get through very, very slowly. For example, this morning, I received a mention at 8:24. My phone showed the notification just fine at that very moment. I left it unread all morning and early afternoon. It's now 1:15 PM as I write this and this despite having my laptop open most of the time and the corebird service running, Corebird only just *now* showed that notification.

**Comment by truckerboy** [comment]
this should quite easily be fixed by adding an extra couple of options in settings:  a) Close to tray (corebird runs in the background as a service) and b) close to exit application (as it does right now). Close to tray should also allow notifications in gnome shell (notifications could be disabled by adding another option).

**Comment by Vistaus** [comment]
@truckerboy There is no tray in GNOME Shell anymore.
https://git.gnome.org/browse/gtk+/commit/?h=gtk-3-22&id=cab7dcde1bef1ea589a9f3c4d8512e59ade8a4a2
https://git.gnome.org/browse/gtk+/commit/?id=d2a8667f835ddce4ac88b9bee143556f9efa93c4

**Comment by truckerboy** [comment]
well there is a tray (deprecated or otherwise) in Gnome 3.20 (which I am on) and it works pretty well but your referred comment makes sense as to why this has not been included in corebird, given it would become throwaway code in the near future.

It would however make a lot of sense to create a gnome shell extension for this (which I am sure is not deprecated) to create a dock and mimic the functionality I outlined in my earlier post.

**Comment by Vistaus** [comment]
A GNOME Shell extension could very well be written and won't be deprecated (it just needs to be kept up-to-date b/w GNOME Shell versions).
